src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -41,11 +42,80 @@ static void add_config(
   describe_configs_source source) {
     result.configs.push_back(describe_configs_resource_result{
       .name = ss::sstring(name),
-      .value = fmt::format("{}", value),
+      .value = ssx::sformat("{}", value),
       .config_source = source,
     });
 }
 
+template<typename T>
+static ss::sstring describe_as_string(const T& t) {
+    return ssx::sformat("{}", t);
+}
+
+template<typename T, typename Func>
+static void add_topic_config(
+  describe_configs_result& result,
+  std::string_view default_name,
+  const T& default_value,
+  std::string_view override_name,
+  const std::optional<T>& override,
+  bool include_synonyms,
+  Func&& describe_f) {
+    describe_configs_source src = override
+                                    ? describe_configs_source::topic
+                                    : describe_configs_source::default_config;
+
+    std::vector<describe_configs_synonym> synonyms;
+    if (include_synonyms) {
+        synonyms.reserve(2);
+        if (override) {
+            synonyms.push_back(describe_configs_synonym{
+              .name = ss::sstring(override_name),
+              .value = describe_f(*override),
+              .source = static_cast<int8_t>(describe_configs_source::topic),
+            });
+        }
+        synonyms.push_back(describe_configs_synonym{
+          .name = ss::sstring(default_name),
+          .value = describe_f(default_value),
+          .source = static_cast<int8_t>(
+            describe_configs_source::default_config),
+        });
+    }
+
+    result.configs.push_back(describe_configs_resource_result{
+      .name = ss::sstring(override_name),
+      .value = describe_f(override.value_or(default_value)),
+      .config_source = src,
+      .synonyms = std::move(synonyms),
+    });
+}
+
+template<typename T>
+static void add_topic_config(
+  describe_configs_result& result,
+  std::string_view default_name,
+  const std::optional<T>& default_value,
+  std::string_view override_name,
+  const tristate<T>& override,
+  bool include_synonyms) {
+    std::optional<ss::sstring> override_value;
+    if (override.is_disabled()) {
```

Comment:
`override` is a reserved keyword, can you use different.

Reply:
this is explicitly allowed in the standard from my reading, but I agree lets change it, even making it plural would be probably be better. GitHub highlights it like a keyword making it harder to read.